app/controllers/user_credentials_controller.rb

```diff
@@ -14,6 +14,26 @@ def index
     end
   end
 
+  def import
+    file = params[:file]
+    content = JSON.parse(file.read)
+    content.each do |hash|
+      current_user.user_credentials << current_user.user_credentials.build(hash.except("id", "created_at", "updated_at", "user_id"))
+    end
+
+    if current_user.save
+      respond_to do |format|
+        format.html { redirect_to user_credentials_path, notice: 'The file was successfully uploaded.' }
+        format.json { render json: @user_credential }
+      end
+    else
+      respond_to do |format|
+        format.html { redirect_to user_credentials_path, notice: 'The file has already been uploaded.'}
```

Comment:
I think you should show a better error message than this.  What sort of errors have you seen when saving?  Are there validation errors on `current_user`?  Or on any of the created credentials?

You could try this:

```ruby
content = JSON.parse(file.read)
new_credentials = content.map do |hash|
 current_user.user_credentials.build(hash.only(%w[credential_name credential_value mode]))
end

if new_credentials.map(&:save).all? { |save_status| save_status } # try to save all of the credentials, and check if any failed
  respond_to do |format|
    format.html { redirect_to user_credentials_path, notice: 'The file was successfully uploaded.' }
  end
else
  respond_to do |format|
    format.html { redirect_to user_credentials_path, notice: 'One or more of the uploaded credentials was not be imported due to error', status: 422 }
  end
end
```

I haven't tried this, but it seems like it would work?